File: e142.java

import java.util.*;
public class e142 
{
    public static void main (String [] args)
    {

        long a, b, c, d, e, f;
        boolean solved = false;
        long result = 0;

        for (long i = 4; !solved; i++) 
        {
            a = i * i;
            for (long j = 3; j < i && !solved; j++) 
            {
                c = j * j;
                f = a - c;
                if (f <= 0 || !IsSquare(f)) continue;
                long kstart = (j % 2 == 1) ? 1 : 2;
                for (long k = kstart; k < j; k += 2) 
                {
                    d = k * k;
                    e = a - d;
                    b = c - e;

                    if (b <= 0 || e <= 0 || !IsSquare(b) || !IsSquare(e)) continue;

                    long x = (d + c) / 2;
                    long y = (e + f) / 2;
                    long z = (c - d) / 2;

                    result = x + y + z;
                    solved = true;
                    break;
                }                    
            }

        }

        System.out.println(result);
    }

    private static boolean IsSquare(long n) 
    { 
        long root=(long)Math.sqrt(n);
        return (root*root==n);
    }


}

File: e142.py
#!/usr/bin/env/ python

from math import sqrt,floor

def is_square(n):
    nroot = floor(sqrt(n))
    return nroot*nroot == n

nsolved = True
i=4

while nsolved:
    a=i*i
    j=3

    while nsolved:
        c=j*j
        f = a-c
        if f<=0 or not is_square(f):
            continue
        kstart = 2
        if j%2==1:
            kstart = 1
        for k in xrange(kstart, j-1, 2):
            d = k*k
            e = a-d
            b = c-e
            if b<=0 or e<=0 or  not is_square(b) or not is_square(e):
                continue
            x = long((d+c)/2)
            y = long((e+f)/2)
            z = long((c-d)/2)

            res = x+y+z
            nsolved = False
            break

        j+=1
    i+=1

print res

